Source/WebCore/dom/ImageOverlay.cpp

```diff
@@ -280,6 +280,9 @@ static Elements updateSubtree(HTMLElement& element, const TextRecognitionResult&
 #endif // ENABLE(MODERN_MEDIA_CONTROLS)
 
     if (RefPtr shadowRoot = element.shadowRoot()) {
+        if (auto* renderer = dynamicDowncast<RenderImage>(element.renderer()))
+            renderer->setHasImageOverlay();
```

Comment:
I'm basically just co-opting "hasImageOverlay" to mean "has shadow content", since that's what it effectively means, and I figured this was preferable to introducing a new method / member variable to RenderImage